src/prefect_server/pollHK.py

```diff
@@ -0,0 +1,148 @@
+from datetime import datetime, timedelta
+from pathlib import Path
+
+import pandas as pd
+from prefect import flow, get_run_logger
+from prefect.runtime import flow_run
+from pydantic import SecretStr
+
+from imap_mag.api.fetch.binary import fetch_binary
+from imap_mag.api.process import process
+from imap_mag.appConfig import manage_config
+from imap_mag.appUtils import HK_APIDS, forceUTCTimeZone, getPacketFromApID
+from imap_mag.DB import Database
+from imap_mag.outputManager import StandardSPDFMetadataProvider
+from prefect_server.constants import CONSTANTS
+from prefect_server.prefectUtils import get_secret_block
+
+
+def convert_ints_to_string(apids: list[int]) -> str:
+    return ",".join(str(apid) for apid in apids)
+
+
+def generate_flow_run_name() -> str:
+    parameters = flow_run.parameters
+    hk_apids = parameters["hk_apids"]
+    start_date = parameters["start_date"] or datetime.today().replace(
+        hour=0, minute=0, second=0, microsecond=0
+    ) - timedelta(days=1)
+    end_date = parameters["end_date"] or start_date + timedelta(days=1)
+
+    apid_text = (
+        f"{convert_ints_to_string(hk_apids)}-ApIDs"
+        if hk_apids != HK_APIDS
+        else "all-HK"
+    )
+
+    return f"Download-{apid_text}-from-{start_date.strftime('%d-%m-%Y')}-to-{end_date.strftime('%d-%m-%Y')}"
+
+
+@flow(
+    name=CONSTANTS.FLOW_NAMES.POLL_HK,
+    log_prints=True,
+    flow_run_name=generate_flow_run_name,
+)
+async def poll_hk_flow(
```

Comment:
We need this logic/code to be testable and probably less dependent to prefect. Prefect server should be a "thin wrapper" around the core business logic and not a critical part of being able to exercise all parts of the system. Lets meet and work out how best to pull this out into it's own thing